Source/WebCore/animation/AnimationTimelinesController.cpp

```diff
@@ -310,42 +311,46 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
         m_cachedCurrentTime = std::nullopt;
 }
 
-void AnimationTimelinesController::registerNamedScrollTimeline(const AtomString& name, Element& source, ScrollAxis axis)
+static Element* originatingElement(Ref<ScrollTimeline> timeline)
 {
-    auto it = m_nameToScrollTimelineMap.find(name);
-    if (it != m_nameToScrollTimelineMap.end()) {
-        auto& existingScrollTimeline = it->value;
-        existingScrollTimeline->setSource(&source);
-        existingScrollTimeline->setAxis(axis);
-    } else {
-        auto newScrollTimeline = ScrollTimeline::create(name, axis);
-        newScrollTimeline->setSource(&source);
-        m_nameToScrollTimelineMap.set(name, WTFMove(newScrollTimeline));
-    }
+    return is<ViewTimeline>(timeline) ? downcast<ViewTimeline>(timeline)->subject() : timeline->source();
```

Comment:
I'm not sure whether it's more efficient, but you could have used a `dynamicDowncast<>` here.

Reply:
Fixed.